src/Interpreters/AsynchronousInsertQueue.cpp

```diff
@@ -218,7 +218,7 @@ AsynchronousInsertQueue::AsynchronousInsertQueue(ContextPtr context_, size_t poo
         dump_by_first_update_threads.emplace_back([this, i] { processBatchDeadlines(i); });
 }
 
-AsynchronousInsertQueue::~AsynchronousInsertQueue()
+void AsynchronousInsertQueue::flushAndShutdown()
```

Comment:
Please comment that an external entity (Context) is responsible for flushing the queue since it's not happening in the destructor anymore.